packages/xchain-ethereum/src/client.ts

```diff
@@ -82,30 +75,23 @@ export default class Client implements XChainClient, EthereumClient {
     vault,
     etherscanApiKey,
   }: ClientParams) {
-    if (phrase && !Crypto.validatePhrase(phrase)) {
-      throw new Error('Invalid Phrase')
-    } else {
-      this._phrase = phrase || Crypto.generatePhrase() // @todo do we need to set a size here?
-      this._network = xchainNetworkToEths(network)
-      this._provider = getDefaultProvider(this._network)
-      this._wallet = ethers.Wallet.fromMnemonic(this._phrase)
-      this._address = this._wallet.address
-      this._vault = vault
-      this._etherscan = new EtherscanProvider(this._network, etherscanApiKey) // for tx history
-
-      // Connects to the ethereum network with it
-      this.changeWallet(this.wallet.connect(this._provider))
-      this.setBlockchairNodeURL(blockchairUrl)
-      this.setBlockchairNodeAPIKey(blockchairNodeApiKey)
+    this.network = xchainNetworkToEths(network)
+    this.provider = getDefaultProvider(this.network)
+    this.etherscan = new EtherscanProvider(this.network, etherscanApiKey) // for tx history
+    this.vault = vault
+    this.setBlockchairNodeURL(blockchairUrl)
+    this.setBlockchairNodeAPIKey(blockchairNodeApiKey)
+
+    if (phrase) {
+      this.setPhrase(phrase)
+      this.changeWallet(this.getWallet().connect(this.provider))
     }
   }
 
   purgeClient(): void {
```

Comment:
Use arrow functions everywhere - see @thatStrangeGuyThorchain [note's about it at Telegram](https://t.me/xchainjs/816):

```
 purgeClient = () => { ... }
```

Pls do the same ^ for all other functions in xchain-eth client.